topological_navigation/scripts/localisation.py

```diff
@@ -153,15 +158,18 @@ def MapCallback(self, msg) :
 
     def Callback(self, msg, item):
         #print item
-        val = getattr(msg, item['field'])
-        if val == item['val'] :
-            if item['name'] not in self.loc_by_topic:
-                self.loc_by_topic.append(item['name'])
-        else:
-            if item['name'] in self.loc_by_topic:
-                self.loc_by_topic.remove(item['name'])
-
-
+        dist = get_distance(self.current_pose, self.previous_pose)
+        if dist>0.10:
+            val = getattr(msg, item['field'])
+            if val == item['val'] :
+                if item['name'] not in self.loc_by_topic:
+                    self.loc_by_topic.append(item['name'])
+                    self.previous_pose = self.current_pose
```

Comment:
no, for the given solution I am storing the position when the status changed if I write it outside the block I would be having the problem described below (see below)